collector/process.go

```diff
@@ -29,20 +30,39 @@ type RlimitUsage struct {
 	Used     uint64 `json:"used"`
 }
 
-func GetProcStats() []ProcessStat {
+func GetProcStats(pidList string) []ProcessStat {
 	stats := make([]ProcessStat, 0)
-	tiServers := []string{"pd-server", "tikv-server", "tidb-server"}
-	for _, procName := range tiServers {
-		proc, err := getProcessesByName(procName)
-		var stat ProcessStat
-		if err != nil {
-			log.Fatal(err)
+	if pidList != "0" {
+		for _, pidStr := range strings.Split(pidList, ",") {
+			pidNum, err := strconv.Atoi(pidStr)
+			if err != nil {
+				log.Fatal(err)
+			}
+			proc, err := getProcessByPID(pidNum)
+			if err != nil {
+				log.Fatal(err)
+			}
+			if proc == nil {
+				return stats
+			}
+			var stat ProcessStat
+			stat.getProcessStat(proc)
+			stats = append(stats, stat)
 		}
-		if proc == nil {
-			continue
+	} else {
+		tiServers := []string{"pd-server", "tikv-server", "tidb-server"}
+		for _, procName := range tiServers {
+			proc, err := getProcessesByName(procName)
+			if err != nil {
+				log.Fatal(err)
+			}
+			if proc == nil {
```

Comment:
It seems inconsistent with the above behavior? [#46](https://github.com/pingcap/tidb-insight/pull/44/files#diff-82a95ed34339398b0ebfe0dc33ef3116R46)

Reply:
BTW, if the loop's logics are same,  move `getProcessByPID`, `getProcessesByName` and itered array as func's params can remove dup.

Reply:
If no process matched, `proc` might be empty, but no error have occurred.